### ISSUE TITLE
letsencrypt certificate renewal

### ISSUE THREAD

**Issue by gitterman** [k9mail/k-9#3571]
I have a mail server with 2 domains signed with letsencrypt certificates.
This works well with K9-mail once I have accepted the certificates on the sending as well as the fetching side.
But every 90 days the certificates get renewed.
Then I need to go into the sending servers settings as well as the fetching servers settings and allow to trust the certificate again.
This involved many clicks in buried sub-menus and as I said needs to be done every 90 days.
Is there a way to add an exception in K9 mail for a certain domain? 
Or could I truncate the clicking process to just a few clicks?
Right now it is a real pain to do this on every android device (Smart Phone, Tablet)

### Expected behavior
allow exception for specific servers and automatically accept new certificates

### Actual behavior
see description above

### Environment
K-9 Mail version: 5.403

Android version:
7.1.1

Account type = IMAP

**Comment by cketti** [comment]
You shouldn't have to manually trust letsencrypt certificates. Maybe the certificate does not include the email server's hostname? Can you post a screenshot of the dialog K-9 Mail displays to accept untrusted certificates?

**Comment by gitterman** [comment]
@cketti: I think you are right that it has to do with the hostname.  
If I accept the certificate it works well for 90 days until it gets renewed.  
My server runs ubuntu 16 and plesk and I have added the certificates using the plesk GUI.


![certificateerror](https://user-images.githubusercontent.com/10762929/44701449-1427d480-aa8f-11e8-8af0-b1b47eeb1dec.jpg)

**Comment by cketti** [comment]
Once we add support for [RFC 6186](https://tools.ietf.org/html/rfc6186) we'll automatically accept a certificate that was issued for the email domain but not the email server's hostname. For now you'll have to add `pop.mandango.de` to the certificate.

**Comment by gitterman** [comment]
I can't add that to the certificate because the certificates are automatically maintained by the PLESK administrative suite.
It is not so much an issue to accept the certificate manually on the receive dialog of K9 mail as it is just 2 clicks. 
The cumbersome part is to manually accept the certificate for the sending side, because the dialog for this is buried deeply. 
I also don't see, why the sending site does not automatically update if the certificate for the receiving site is acknowledged.
On my PC I use thunderbird and there I do only have to accept the renewed certificate once.

**Comment by cketti** [comment]
`pop.mandango.de` and `mandango.de` resolve to the same IP. That means you could just use `mandango.de` as hostname for the incoming and outgoing server and you wouldn't have to manually accept the certificate.

**Comment by rantingnerd** [comment]
I have what looks like this problem as well (see attachments).  I use mail.interstel.net for IMAP, and mail.interstel.net is in the certificate as an alternative name.  Every time the cert updates, I have to re-click through and accept the certificate.  I run stock Android (11) straight from Google (no OME intermediary).   

Is there a timeline for RFC 6186 implementation?  I ran across https://github.com/k9mail/k-9/issues/2361 (to remove the ability to accept a certificate) and would hate for that to get implemented before the RFC 6186 fix.

Thanks --

![K-9-Cert-error-1-20201011-164124](https://user-images.githubusercontent.com/33764072/95689874-e9068d80-0be1-11eb-902e-b469307c4267.png)
![K-9-Cert-error-2-20201011-164132](https://user-images.githubusercontent.com/33764072/95689876-e9068d80-0be1-11eb-983d-fd98ac5909c3.png)

**Comment by cketti** [comment]
@rantingnerd: Please use the [forum](https://forum.k9mail.app/) for support requests.
Your issue seems to be that you don't have the necessary root certificate installed. Not that the host names don't match.

**Comment by rantingnerd** [comment]
Sorry for the mis-fire, and thanks for the pointer and diagnosis!

**Comment by robo45h** [comment]
> @rantingnerd: Please use the [forum](https://forum.k9mail.app/) for support requests.
> Your issue seems to be that you don't have the necessary root certificate installed. Not that the host names don't match.

I had this issue myself -- the error was about the "trust anchor" (root certificate). However, as far as I can tell, I **did** have this installed in a way that satisfied Chrome and Mozilla and Roundcube (web email) and Thunderbird email (Windows) talking to Dovecot IMAP. But I noticed that the Dovecot configuration only allows specification of one ssl_cert and an ssl_key. For me these were domain.cer and domain.key. However, in my web server ssl directory there was also a fullchain.cer file. I saw two potential changes: changing Dovecot to point to fullchain.cer for the ssl_cert, or just replacing my domain.cer file with a copy of the fullchain.cer file. Probably the more limited change to Dovecot would have worked, but I went the other way, and that did resolve my problem. Posting this here to help others, as I could not find this question answered in my Google searches.
